render/container.go

```diff
@@ -264,6 +264,11 @@ func MapContainer2IP(m report.Node) []string {
 			if !ok {
 				continue
 			}
+			// loopback addresses are shared among all namespaces
+			// so we can't use them to attribute connections to a container
+			if report.IsLoopback(addr) {
+				continue
+			}
```

Comment:
To elaborate a bit further, it will still work when PIDs are provided. For instance, when reproducing the bug with the method suggested by @rade on https://github.com/weaveworks/scope/issues/1733#issuecomment-236395353 I get:

<img width="1150" alt="screen shot 2016-08-10 at 11 38 42 am" src="https://cloud.githubusercontent.com/assets/2362916/17555881/32d3b468-5f09-11e6-9f75-030129291722.png">

and after the fix:

<img width="1149" alt="screen shot 2016-08-10 at 11 31 57 am" src="https://cloud.githubusercontent.com/assets/2362916/17555905/49b39c48-5f09-11e6-8c7a-8cf8927ce728.png">

Note how loops are still detected when using persistent connections.